Detectors/TRD/calibration/src/DCSProcessor.cxx

```diff
@@ -0,0 +1,246 @@
+      }
+    }
+  }
+  return 0;
+}
+
+//______________________________________________________________________
+
+int DCSProcessor::processFlags(const uint64_t flags, const char* pid)
+{
+
+  // function to process the flag. the return code zero means that all is fine.
+  // anything else means that there was an issue
+
+  // for now, I don't know how to use the flags, so I do nothing
+
+  if (flags & DataPointValue::KEEP_ALIVE_FLAG) {
+    LOG(info) << "KEEP_ALIVE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::END_FLAG) {
+    LOG(info) << "END_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::FBI_FLAG) {
+    LOG(info) << "FBI_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::NEW_FLAG) {
+    LOG(info) << "NEW_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::DIRTY_FLAG) {
+    LOG(info) << "DIRTY_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::TURN_FLAG) {
+    LOG(info) << "TURN_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::WRITE_FLAG) {
+    LOG(info) << "WRITE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::READ_FLAG) {
+    LOG(info) << "READ_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::OVERWRITE_FLAG) {
+    LOG(info) << "OVERWRITE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::VICTIM_FLAG) {
+    LOG(info) << "VICTIM_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::DIM_ERROR_FLAG) {
+    LOG(info) << "DIM_ERROR_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_DPID_FLAG) {
+    LOG(info) << "BAD_DPID_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_FLAGS_FLAG) {
+    LOG(info) << "BAD_FLAGS_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_TIMESTAMP_FLAG) {
+    LOG(info) << "BAD_TIMESTAMP_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_PAYLOAD_FLAG) {
+    LOG(info) << "BAD_PAYLOAD_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_FBI_FLAG) {
+    LOG(info) << "BAD_FBI_FLAG active for DP " << pid;
+  }
+
+  return 0;
+}
+
+//______________________________________________________________________
+
+void DCSProcessor::updateDPsCCDB()
+{
+  // here we create the object to then be sent to CCDB
+  LOG(info) << "Finalizing";
+  // TODO I do not know why this converter works, but I don't want to go deeper into the DCS code at the moment
+  union Converter {
+    uint64_t raw_data;
+    double double_value;
+  } converter;
+
+  for (const auto& it : mPids) {
+    const auto& type = it.first.get_type();
+    if (type == o2::dcs::RAW_DOUBLE) {
+      auto& trddcs = mTRDDCS[it.first];
+      if (it.second == true) { // we processed the DP at least 1x
+        auto& dpVect = mDpsDoublesmap[it.first];
+        for (const auto& dpVal : dpVect) {
+          converter.raw_data = dpVal.payload_pt1;
+          double value = converter.double_value;
+          if (trddcs.nPoints == 0) {
```

Comment:
Minor style issue: this could be moved to a method of TRDDCSMinMaxMeanInfo

Reply:
Done